src/equations/shallow_water_1d.jl

```diff
@@ -408,6 +441,30 @@ Further details on this hydrostatic reconstruction and its motivation can be fou
   return u_ll_star, u_rr_star
 end
 
+@inline function  hydrostatic_reconstruction_chen_noelle(u_ll, u_rr, equations::ShallowWaterEquations1D)
```

Comment:
Add a docstring here to breifly describe their HR procedure. Also include a reference and DOI to the Chen & Noelle paper.